api/backend/duckdb.py

```diff
@@ -3,34 +3,49 @@
 
 from loguru import logger
 
-schema_lineitem = """CREATE TABLE IF NOT EXISTS lineitem(
-            l_orderkey INTEGER NOT NULL, 
-            l_partkey INTEGER NOT NULL, 
-            l_suppkey INTEGER NOT NULL, 
-            l_linenumber INTEGER NOT NULL, 
-            l_quantity INTEGER NOT NULL, 
-            l_extendedprice DECIMAL(15,2) NOT NULL, 
-            l_discount DECIMAL(15,2) NOT NULL, 
-            l_tax DECIMAL(15,2) NOT NULL, 
-            l_returnflag VARCHAR NOT NULL, 
-            l_linestatus VARCHAR NOT NULL, 
-            l_shipdate DATE NOT NULL, 
-            l_commitdate DATE NOT NULL, 
-            l_receiptdate DATE NOT NULL, 
-            l_shipinstruct VARCHAR NOT NULL, 
-            l_shipmode VARCHAR NOT NULL, 
-            l_comment VARCHAR NOT NULL);"""
 
+class DuckDBConnection:
+    def __init__(self):
+        self.conn_pool = []
 
-def ConnectDuckDB():
-    con = duckdb.connect()
-    con.install_extension("substrait")
-    con.load_extension("substrait")
-    con.install_extension("tpch")
-    con.load_extension("tpch")
-    con.execute(query=schema_lineitem)
-    logger.success("DuckDb initialized successfully")
-    return con
+        query_lineitem = """CREATE TABLE IF NOT EXISTS lineitem(
+                        l_orderkey INTEGER NOT NULL, 
+                        l_partkey INTEGER NOT NULL, 
+                        l_suppkey INTEGER NOT NULL, 
+                        l_linenumber INTEGER NOT NULL, 
+                        l_quantity INTEGER NOT NULL, 
+                        l_extendedprice DECIMAL(15,2) NOT NULL, 
+                        l_discount DECIMAL(15,2) NOT NULL, 
+                        l_tax DECIMAL(15,2) NOT NULL, 
+                        l_returnflag VARCHAR NOT NULL, 
+                        l_linestatus VARCHAR NOT NULL, 
+                        l_shipdate DATE NOT NULL, 
+                        l_commitdate DATE NOT NULL, 
+                        l_receiptdate DATE NOT NULL, 
+                        l_shipinstruct VARCHAR NOT NULL, 
+                        l_shipmode VARCHAR NOT NULL, 
+                        l_comment VARCHAR NOT NULL);"""
+        conn = duckdb.connect("duck.db")
+        conn.execute(query=query_lineitem)
+
+        for i in range(5):
+            conn = duckdb.connect("duck.db")
+            conn.install_extension("substrait")
+            conn.load_extension("substrait")
+            self.conn_pool.append(conn)
+
+    def check_pool(self):
+        if len(self.conn_pool) == 0:
+            for i in range(5):
+                conn = duckdb.connect("duck.db")
+                conn.install_extension("substrait")
+                conn.load_extension("substrait")
+                self.conn_pool.append(conn)
+
+    def initialize(self):
```

Comment:
does a name like get_connection make more sense here?

Reply:
yep, that's right, made the change, thanks!